components/servo/lib.rs

```diff
@@ -331,6 +331,11 @@ where
                 webrender_gl.clone(),
                 render_notifier,
                 webrender::WebRenderOptions {
+                    // We force the use of optimized shaders here because rendering is broken
+                    // on Android emulators with unoptimized shaders. This is due to a known
+                    // issue in the emulator's OpenGL emulation layer.
+                    // See: https://github.com/servo/servo/issues/31726
+                    use_optimized_shaders: true,
```

Comment:
I've manually tested the NixOS and Android builds but I'm no sure if this change could potentially break other platforms. WebRender does have fallback logic to use unoptimized versions if the optimized version is not present in the build. However, let me know if we should instead enable this only for android i.e `use_optimized_shaders: cfg!(target_os = "android")` instead.